### DISCUSSION TITLE
List of all html elements (and maybe attributes/CSS classes) used by Katex

### DISCUSSION THREAD

**Discussion by TruePath** [KaTeX/KaTeX#3489] [Q&A] [answered]
As noted in the [security](https://katex.org/docs/security.html) page in the docs it's a good idea to take your katex output and put it through a sanitizer but to do that I need to know the complete list of tags (and ideally attributes and classes) produced by Katex.  Is there someplace I can find a complete list (ideally both for mathml and html output)?  Thanks!  (Sorry if this is addressed somewhere but I didn't find it when I searched).

---

P.S. I'm open to other suggestions but I'm using Katex as part of kramdown on the server and while I'm planning to strip all html tags out of the markdown before I process it I'm worried that it will be possible to cleverly use markdown elements to create tags in the output that weren't in the input (e.g. if it removes an intervening newline or escape somehow).

**Comment by edemaine** [comment, marked as answer]
One list of elements and classes is here: https://github.com/edemaine/coauthor/blob/main/lib/sanitize.coffee

I agree it'd be nice to have an autogenerated list, or perhaps even export it from KaTeX.
